desc/protoparse/registered_descriptor_test.go

```diff
@@ -0,0 +1,42 @@
+package protoparse
+
+import "testing"
+
+func TestGetRegisteredDescriptors(t *testing.T) {
```

Comment:
I think we instead wants tests of the parser _using_ the new field, to make sure they are correctly used. You could do this via test like so:

1. Compile a proto source.
2. Parse a second proto source that depends on the first. Pass the resulting descriptor from the first step in the newly created field. Make sure this parse would fail if it tried to actually load source for the file. It should still correctly parse.
3. Same as step 2, except that also supply a _different_ version of the dependency as source (available via the file accessor), and then make sure that alternate version is not actually used (that the provided descriptor was used instead of trying to parse source).

Reply:
Added a test with (1) and (2).

Holding off on (3) for the outcome of our other discussion.

Reply:
Per other decisions, the test should verify that the source file version was used, not the descriptor that was put into `DependencyDescriptors`.

Reply:
Done.